pkg/backend/display/progress_test.go

```diff
@@ -120,3 +124,116 @@ func TestProgressEvents(t *testing.T) {
 		})
 	}
 }
+
+// The following test checks that the status display elements have retain on delete details added.
+func TestStatusDisplayFlags(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name         string
+		stepOp       display.StepOp
+		shouldRetain bool
+	}{
+		// Should display `retain`.
+		{"delete", deploy.OpDelete, true},
+		{"replace", deploy.OpReplace, true},
+		{"create-replacement", deploy.OpCreateReplacement, true},
+		{"delete-replaced", deploy.OpDeleteReplaced, true},
+
+		// Should be unaffected.
+		{"same", deploy.OpSame, false},
+		{"create", deploy.OpCreate, false},
+		{"update", deploy.OpUpdate, false},
+		{"read", deploy.OpRead, false},
+		{"read-replacement", deploy.OpReadReplacement, false},
+		{"refresh", deploy.OpRefresh, false},
+		{"discard", deploy.OpReadDiscard, false},
+		{"discard-replaced", deploy.OpDiscardReplaced, false},
+		{"import", deploy.OpImport, false},
+		{"import-replacement", deploy.OpImportReplacement, false},
+
+		// "remove-pending-replace" is not a valid step operation.
+		// {"remove-pending-replace", deploy.OpRemovePendingReplace, false},
+	}
+
+	for _, test := range testCases {
+		tt := test
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
The name should especially  match here because otherwise, there's a risk of accidentally using the wrong value: e.g. `test` instead of `tt`.